### ISSUE TITLE
Allow custom statuses from mods for units

### ISSUE THREAD

**Issue by RealityMachina** [X2CommunityCore/X2WOTCCommunityHighlander#322]
The specific function to change would be `GetStatusStringsSeparate` in `XComGameState_Unit`, which conveniently is the one stop place the game has for knowing what kind of status string to deliver.

By default it just checks projects and covert actions, but an event trigger for a tuple to come back with a string when applicable would allow for mods with unique strategy functions for units to give proper statuses of units interacting with them.

**Comment by Xymanek** [comment]
Maybe a new `ESoldierStatus` entirely? Something like `eStatus_AwayFromAvenger` that strategy mods can then use as they want?

**Comment by Xymanek** [comment]
The intend of my initial idea was to not confuse other mods and game systems which may attempt to look for the current covert op. However thinking about this more, maybe we should take this a step further? Make something like `CHItemSlot` but for soldier status. Some idea for callbacks it will have:

* `GetStatusString`
* `GetTimeValue`
* `CanVisitInArmoury` (thinking of LW2 advisors)
* `CanChangeLoadout` (thinking of LW2 advisors again but during retals in that region)
* `CanGainXP` (for the all soldiers gain XP mod)

Pretty sure there can be other uses. As for how it can be implemented, we can take inspiration from Firaxis:

```
//@TODO - rmcfall/jbouscher - Refactor these enums? IE. make location a state object reference to something? and make status based on the location?
//*******************************************
enum ESoldierStatus
```

**Comment by Xymanek** [comment]
For the record: I spoke to robo on discord and he gave a very good reason to keep using `eStatus_CovertAction` - existing code/mods uses it to distinguish units that are not on avenger. So I scrap my above suggestions for sake of simplicity

**Comment by robojumper** [comment]
Reopening because this needed bit of a redesign and the rewrite PR ended up more intrusive than initially expected.

**Comment by Xymanek** [comment]
Implemented in 1.19
